### ISSUE TITLE
BroadcastReceiver Issues in LAMP-kotlin

### ISSUE THREAD

**Issue by MattMatt27** [BIDMCDigitalPsychiatry/LAMP-platform#816]
**Describe the bug**
LAMP_kotlin.sensor_core.Screen.onStartCommand is failing with the following Java security exception on Android 14 devices.
```
Caused by java.lang.SecurityException
digital.lamp.mindlamp: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
```
The issue appears to be new security requirements for the registerReceiver command explained here: https://developer.android.com/about/versions/14/behavior-changes-14#runtime-receivers-exported

The following line is the one that is currently failing: https://github.com/BIDMCDigitalPsychiatry/LAMP-kotlin/blob/6c0d9692a47ef7a639f231d4d60de6dd7e9ec688/lamp_kotlin/src/main/java/digital/lamp/lamp_kotlin/sensor_core/Screen.kt#L86

**To Reproduce**
- Install lamp on an Android 14 device
- Allow data collection
- Screen State data should fail and be visible in Firebase Crashlytics

**Expected behavior**
The error should clear and screen state data collection should resume for Android 14 users

**Smartphone (please complete the following information):**
 - Device: Mostly Samsungs, some Pixels
 - OS: Android 14

**Additional context**
There may be other registerReceiver commands in LAMP-kotlin that require updates. A quick search turned up a half dozen instances of the command without the new flag, but I am unsure if all of them actually need the update because supposedly system broadcasts do not require it. Also if we have any receivers defined in `AndroidManifest.xml` those will likely need updates as well. That said, for now the Screen script seems to be the only one that is failing.

**Comment by sijitg** [comment]
We are checking it. We shall update ASP

**Comment by sijitg** [comment]
We will update  LAMP-kotlin and app staging release pointing to new library version today. Thanks [MattMatt27](https://github.com/MattMatt27)  for the debug info.

**Comment by sijitg** [comment]
We merged the fix.  LAMP_kotlin library version 2024.07.12 is released.  mindLamp app version 2024.04.07 Open testing (staging) build  is submitted for Google review.

**Comment by sijitg** [comment]
Production version for this issue fix is live in play store (Version 2024.07.12)

**Comment by MattMatt27** [comment]
Confirmed in the Firebase Crashlytics Dashboard that this issue is resolved
